jupyter_events/logger.py

```diff
@@ -107,94 +79,48 @@ def get_handlers():
-
-        Supports both JSON & YAML files.
+    def register_event_schema(self, schema: Union[dict, str]):
+        """Register this schema with the schema registry.
 
-        Parameters
-        ----------
-        filename: str, path object or file-like object
-            Path to the schema file or a file object to register.
+        Get this registered schema using the EventLogger.schema.get() method.
         """
-        # Just use YAML loader for everything, since all valid JSON is valid YAML
+        self.schemas.register(schema)
 
-        # check if input is a file-like object
-        if hasattr(filename, "read") and hasattr(filename, "write"):
-            self.register_schema(yaml.load(filename))
-        else:
-            with open(filename) as f:
-                self.register_schema(yaml.load(f))
-
-    def register_schema(self, schema):
-        """
-        Register a given JSON Schema with this event emitter
+    def register_event_schema_file(self, schema_file: str):
+        """Register this schema with the schema registry.
 
-        'version' and '$id' are required fields.
+        Get this registered schema using the EventLogger.schema.get() method.
         """
-        # Check if our schema itself is valid
-        # This throws an exception if it isn't valid
-        JSONSchemaValidator.check_schema(schema)
-
-        # Check that the properties we require are present
-        required_schema_fields = {"$id", "version", "properties"}
-        for rsf in required_schema_fields:
-            if rsf not in schema:
-                raise ValueError(f"{rsf} is required in schema specification")
-
-        if (schema["$id"], schema["version"]) in self.schemas:
-            raise ValueError(
-                "Schema {} version {} has already been registered.".format(
-                    schema["$id"], schema["version"]
-                )
-            )
-
-        for p, attrs in schema["properties"].items():
-            if p.startswith("__"):
-                raise ValueError(
-                    "Schema {} has properties beginning with __, which is not allowed"
-                )
-
-            # Validate "categories" property in proposed schema.
-            try:
-                cats = attrs["categories"]
-                # Categories must be a list.
-                if not isinstance(cats, list):
-                    raise ValueError(
-                        'The "categories" field in a registered schemas must be a list.'
-                    )
-            except KeyError:
-                raise KeyError(
-                    'All properties must have a "categories" field that describes '
-                    'the type of data being collected. The "{}" property does not '
-                    "have a category field.".format(p)
-                )
-
-        self.schemas[(schema["$id"], schema["version"])] = schema
+        self.schemas.register_from_file(schema_file)
```

Comment:
Could this read in the file and perform that kind of validation here, then call `self.schemas.register(schema)`?  This would prevent the need for implementing `self.schemas.register_from_file(schema_file)` and an additional layer of support.

Reply:
Good idea. I actually just consolidated these methods. 

The API now has a single method, `register_event_schema`, which can take a string, pathlib object, or schema object. It handles the loading under the hood depending on the datatype.